flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/nonha/embedded/EmbeddedJobResultStore.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.highavailability.nonha.embedded;
+
+import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.highavailability.JobResultEntry;
+import org.apache.flink.runtime.highavailability.JobResultStore;
+import org.apache.flink.runtime.jobmaster.JobResult;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+/**
+ * An implementation of the {@link JobResultStore} which only persists the data to an in-memory map.
+ */
+public class EmbeddedJobResultStore implements JobResultStore {
```

Comment:
@autophagy @XComp Could you please reason about which threads will be accessing the JRS? Is it OK to have a non-thread safe implementation (eg. are we guaranteed that only main thread accesses it)?

Reply:
@dmvk @XComp and I paired on this and determined that, due to the JRS being initalized along with the jobgraph, it will be accessed in the ioExecutor threadpool - not just the main thread. I will use a threadsafe implementation here.